scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -1025,6 +1028,9 @@ def unit_addr(self) -> Optional[int]:
         try:
             addr = int(self.name.split("@", 1)[1], 16)
         except ValueError:
+            # Non-hex addresses are ok for PCI devices (e.g. "pcie@1,0")
+            if self.is_pci_device:
+                return None
```

Comment:
Please update the class docstring to reflect this additional scenario

Reply:
@mbolivar-ampere thanks, I modified this slightly to just always return `None` for PCI devices.

Possibly further validation could be added to PCI devices, although the device tree compiler already outputs warnings if the node name does not align with the reg node (see: `check_pci_device_reg` and `check_pci_device_bus_num`):
https://github.com/torvalds/linux/blob/master/scripts/dtc/checks.c#L903